### PR TITLE
ci: add integration testing for etcd-client

### LINKED ISSUE #9093: Run etcd-client tests as a part of Tarantool integration testing
We have a Lua module `etcd-client` that implements a Lua API with access to `etcd` key-value storage.
`etcd-client` uses Tarantool's built-in http client and Tarantool's http client based on Curl library. 
It would be nice to add `etcd-client` tests as a part of Tarantool's integration testing. This will guarantee that curl bumps and changes in http client will not break `etcd-client` behavior.

Latest breakage: #9086, upstream https://github.com/curl/curl/pull/11795 (NOTE: fix was committed without regression test, issue could happen again).

### PR DESCRIPTION
**Pull request by ylobankov**
Add workflow job for integration testing of the etcd-client module.

Closes #9093

### CODE REVIEW

**Review by ochaplashkin — APPROVED** [review]

**Review by foxzi — APPROVED** [review]
OK

**Comment by ylobankov** [comment]
Screenshot 2024-03-07 at 14 17 26

File: .github/workflows/etcd_integration.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        type: string
+
+jobs:
+  run_tests:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        etcd-version: [ 'v3.4.30', 'v3.5.12' ]
```
Comment (ligurio) [inline .github/workflows/etcd_integration.yml:19]: I would add a link https://endoflife.date/etcd to the comment.
Feel free to ignore.

**Review by ligurio — APPROVED** [review]
Yaroslav, thanks for the patch! LGTM with a minor comment.

File: .github/workflows/etcd_integration.yml (hunk 1 above)
Comment (ylobankov) [reply .github/workflows/etcd_integration.yml:20]: Fixed.

**Pull request merged**